src/compiler.rs

```diff
@@ -224,7 +224,11 @@ impl AmberCompiler {
             let dir_path = {
                 let file_dir = dep_path.strip_prefix(&base_dir).unwrap();
                 let parent = file_dir.parent().unwrap().display();
-                format!("{}/{output}/{}", base_dir.to_string_lossy(), parent)
+                if !output.is_empty() {
+                    output.clone()
+                } else {
+                    format!("{}/{output}/{}", base_dir.to_string_lossy(), parent)
+                }
```

Comment:
This string will never be empty, because when not provided, then it'll be defined as `"docs/"`. Thus we can simplify this snippet and leave it as it was before or even use `PathBuf` api to join the paths:

```rs
use std::path::PathBuf;

fn main() {
    // Declare the PathBuf variables
    let mut a = PathBuf::from("a");
    let b = PathBuf::from("b");
    let c = PathBuf::from("c");

    // Join the paths together
    a.push(b);
    a.push(c);

    // Print the final result
    println!("{}", a.display());
}
```

Reply:
I tried to do something else with path switching but my rust skills are very crappy and with `&str` `String` and `pathbuf` I was getting crazy...

Reply:
in my idea the solution was to let to the user what is the path he prefer and also the folder where is the documentation

Reply:
I understand... but then you still need to join the parent like this: `{output}/{parent}`

Reply:
I have no idea how to do that in Rust about path detection.
@hdwalters can you help for this Rust line?

Reply:
See direct message on Discord server.

Reply:
But to summarise, I already modified this code on a different branch; see https://github.com/hdwalters/amber/blob/460c612542f5618ee912330e76e134e730975b0c/src/compiler.rs#L222.

Reply:
```
// Save to file; replace the base directory if the output
// path is absolute, otherwise append the output path.
let dir_path = {
    let file_path = dep_path.strip_prefix(&base_dir).unwrap();
    let file_dir = file_path.parent().unwrap();
    base_dir.join(&output).join(file_dir)
};
```

Reply:
The point is that `PathBuf::join()` automatically detects if the joined path is relative or absolute.

Reply:
Remove that line as @hdwalters has an open PR that include the same fix